lib/kubernetes-deploy/resource_discovery.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module KubernetesDeploy
+  class ResourceDiscovery
+    def initialize(namespace:, context:, logger:, namespace_tags:)
+      @namespace = namespace
+      @context = context
+      @logger = logger
+      @namespace_tags = namespace_tags
+    end
+
+    def crds(sync_mediator)
```

Comment:
Are you worried about the sync mediator's cache? I'd assumed we wanted to discourage direct calls to kubeclient

Reply:
Not really worried about the cache, since we clear that at the beginning of every sync cycle. We definitely want to discourage direct calls to kubeclient from resource classes--they should only be making calls during the sync cycle, and therefore should only need the sync mediator (and can all benefit from its cache if they do). But what we're doing here has nothing to do with the sync cycle and the mediator gives us no benefit. EjsonSecretProvisioner is an example of another piece of independent functionality that also takes care of its own calls.